src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestGHEventSubscriber.java

```diff
@@ -80,50 +102,217 @@ protected boolean isApplicable(@Nullable Job<?, ?> project) {
+
+                        @Override
+                        public void run() {
+                            SCMHeadEvent.fireNow(new SCMHeadEventImpl(SCMEvent.Type.CREATED, p, changedRepository));
+                        }
+                    }, 5, TimeUnit.SECONDS);
+                } else if ("reopened".equals(action) || "synchronize".equals(action)) {
+                    Timer.get().schedule(new Runnable() {
+                        @Override
+                        public void run() {
+                            SCMHeadEvent.fireNow(new SCMHeadEventImpl(SCMEvent.Type.UPDATED, p, changedRepository));
+                        }
+                    }, 5, TimeUnit.SECONDS);
+                } else if ("closed".equals(action)) {
+                    Timer.get().schedule(new Runnable() {
+                        @Override
+                        public void run() {
+                            SCMHeadEvent.fireNow(new SCMHeadEventImpl(SCMEvent.Type.REMOVED, p, changedRepository));
+                        }
+                    }, 5, TimeUnit.SECONDS);
+                }
             }
-            // Delaying the indexing for some seconds to avoid GitHub cache
-            Timer.get().schedule(new Runnable() {
-                @Override public void run() {
-                    ACL.impersonate(ACL.SYSTEM, new Runnable() {
-                        @Override public void run() {
-                            boolean found = false;
-                            for (final SCMSourceOwner owner : SCMSourceOwners.all()) {
-                                for (SCMSource source : owner.getSCMSources()) {
-                                    if (source instanceof GitHubSCMSource) {
-                                        GitHubSCMSource gitHubSCMSource = (GitHubSCMSource) source;
-                                        if (gitHubSCMSource.getRepoOwner().equals(changedRepository.getUserName()) &&
-                                                gitHubSCMSource.getRepository().equals(changedRepository.getRepositoryName())) {
-                                            owner.onSCMSourceUpdated(gitHubSCMSource);
-                                            LOGGER.log(Level.FINE, "PR event on {0}:{1}/{2} forwarded to {3}", new Object[] {changedRepository.getHost(), changedRepository.getUserName(), changedRepository.getRepositoryName(), owner.getFullName()});
-                                            found = true;
-                                        }
-                                    }
-                                }
+
+        } catch (IOException e) {
+            LogRecord lr = new LogRecord(Level.WARNING, "Could not parse {0} event with payload: {1}");
+            lr.setParameters(new Object[]{event, payload});
+            lr.setThrown(e);
+            LOGGER.log(lr);
+        }
+    }
+
+    private static class SCMHeadEventImpl extends SCMHeadEvent<GHEventPayload.PullRequest> {
+        @EventData
```

Comment:
I feel like I've missed something, what does this annotation mean? It wasn't on `NewSCMSourceEvent`